### PR TITLE
[FIX] website: only one website on dashboard if no multi-website

### PR DESCRIPTION
**Pull request by rdeodoo**
If multi-website is disabled, we still show all the websites on the website
analytics dashboard.

This commit will only show one website if multi-website is disabled

### CODE REVIEW

File: addons/website/controllers/backend.py [hunk 1]
```diff
@@ -25,7 +25,8 @@ def fetch_dashboard_data(self, website_id, date_from, date_to):
         }
 
         current_website = website_id and Website.browse(website_id) or Website.get_current_website()
-        dashboard_data['websites'] = request.env['website'].search_read([], ['id', 'name'])
+        multi_website = request.env.user.has_group('website.group_multi_website')
+        dashboard_data['websites'] = multi_website and request.env['website'].search_read([], ['id', 'name']) or current_website.read(['id', 'name'])
```
[the remark this answers is not in the document]
Comment (rdeodoo) [reply addons/website/controllers/backend.py]: As it was:
<pre>
multi_website and request.env['website'].search_read([], ['id', 'name']) or current_website.read(['id', 'name'])
</pre>
Or as you suggest but with `search()` and not `search_read()`
<pre>
(multi_website and request.env['website'].<b>search([])</b> or current_website).read(['id', 'name'])
</pre>

I pushed the second one, robodoo r+ if you are ok

**Comment by sswapnesh** [comment]
Hello @JKE-be @rdeodoo  I think we should also check the Website settings If Multi Website is enabled or not.

What do you think on this ?

**Comment by JKE-be** [comment]
It is the purpose of this commit ...

I don’t understand your remark

**Comment by sswapnesh** [comment]
Here  (`request.env.user.has_group('website.group_multi_website')`) we are just checking if user has rights of Multi website or not we are not checking if the feature is enabled or not.
screenshot 26
May be I am wrong here if it is already there.

**Comment by JKE-be** [comment]
Yes but when you check the box, it add the groups and vice versa

So it is ok

**Comment by JKE-be** [comment]
robodoo r+

**Comment by sswapnesh** [comment]
> Yes but when you check the box, it add the groups and vice versa
> 
> So it is ok

Are you Sure ?

Because there can be a case Where  `Multi Website` feature is enabled but all user have not rights of `Multi Website` .

**Comment by sswapnesh** [comment]
BTW, Adding `?fw=` in url allows you to visit multiple websites even Multi website is disabled.

**Comment by sswapnesh** [comment]
Still It has Issue
Give Multi website rights
screenshot 28

Disable that feature
screenshot 27
Go to Dashboard
screenshot 29

**Comment by rdeodoo** [comment]
> Because there can be a case Where `Multi Website` feature is enabled but all user have not rights of `Multi Website` .
Then the user will only see his current website with this PR which is good

> BTW, Adding ?fw= in url allows you to visit multiple websites even Multi website is disabled.

Multi-Website is enabled automatically when you create your second website. So `?fw=` would only shows you other websites if you disable yourself the settings and still have more than one website.
The normal use case would be to have one website, then `?fw=` would do nothing as there is only one website.
Then you might want to activate multi website, in that case you still only have one website.
Or you would create a second one that would automatically activate multi website. Then `?fw=` would do its job.

The only flows to be 'weird' would be to create a second website then disable multi website then use `?fw=` to navigate to second site.. I am not sure what would be the problem tho.. As the user should himself adds the query string to the URL since there is no way to get that URL from Odoo if you disable multi websites

**Comment by rdeodoo** [comment]
> Still It has Issue
> Give Multi website rights
> screenshot 28
> 
> Disable that feature
> screenshot 27
> Go to Dashboard
> screenshot 29

That flow is impossible, if multi website is already enabled (it is right since you disable it after in your flow?) then the user already has multi website?
Disabling multi website will then remove the multi website group for that user and he will onle see the current website.
Note that this is the same case if you disable then enable the group on the user before disabling multi website..

>  and what will be the case when there are Two websites and user has not access to other website and >  add fw url? (When I Tested I was able to add fw and access other website w/o login).

Please open another issue if you see a bug as this has nothing to do with this PR

**Pull request closed without merging**